cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java

```diff
@@ -151,23 +151,20 @@ public Secret cruiseControlSecret() {
             isMaintenanceTimeWindowsSatisfied);
     }
 
-    public Map<String, CertAndKey> generateZkCerts(Kafka kafka, boolean isMaintenanceTimeWindowsSatisfied) throws IOException {
-        String cluster = kafka.getMetadata().getName();
-        String namespace = kafka.getMetadata().getNamespace();
-
-        DnsNameGenerator zkDnsGenerator = DnsNameGenerator.of(namespace, KafkaResources.zookeeperServiceName(cluster));
-        DnsNameGenerator zkHeadlessDnsGenerator = DnsNameGenerator.of(namespace, KafkaResources.zookeeperHeadlessServiceName(cluster));
+    public Map<String, CertAndKey> generateZkCerts(String namespace, String kafkaName, int replicas, boolean isMaintenanceTimeWindowsSatisfied) throws IOException {
```

Comment:
why did you prefer to pass `namespace` and `kafkaName` instead the current only `Kafka` instance?
are we using `kafkaName` in any other places? or just `name` or `clusterName`?

Reply:
Because the Kafka instance is _big_ and it would need to be stored through the reconciliation (today we use it only on the beginning to create the model, so there was not other reason to keep it in the state). And at the same time, it turned out it wasn't really needed.